### PR TITLE
1061 test properties validation

### PR DESCRIPTION
**Pull request by splekhanov**
#1061 
Properties validation mechanism and 'timeout.wait.element' test

### CODE REVIEW

File: jdi-light/src/main/java/com/epam/jdi/light/common/PropertyNames.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.epam.jdi.light.common;
+
+public class PropertyNames {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/common/PropertyNames.java]: Codacy Issue found: [The utility class name 'PropertyNames' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  4
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  4
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

File: jdi-light/src/main/java/com/epam/jdi/light/settings/PropertyNames.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.epam.jdi.light.settings;
+
+public class PropertyNames {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/settings/PropertyNames.java]: Codacy Issue found: [The utility class name 'PropertyNames' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

Comment (mf-code) [reply jdi-light/src/main/java/com/epam/jdi/light/settings/PropertyNames.java]: need to supress (you can ask help at @RomE-HK )

File: jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Test(expectedExceptions = {RuntimeException.class}, dataProvider = "negativeData",
+            expectedExceptionsMessageRegExp = ".*Value of parameter '" + TIMEOUT_WAIT_ELEMENT + "' is not valid." +
+                    " See example: https://jdi-docs.github.io/jdi-light/#driver-settings.*")
+    public void negativeTest(String value) {
+        Properties properties = new Properties();
+        properties.setProperty(TIMEOUT_WAIT_ELEMENT, value);
+        validateProperties(properties);
+        Assert.fail("Value " + value + " should not be valid for this test.");
```
Comment (mf-code) [inline jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java]: `fail("Value " + value + " should not be valid for this test.");` (add static import for fail)

File: jdi-light/src/main/java/com/epam/jdi/light/settings/PropertyNames.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/settings/PropertyNames.java]: Codacy Issue found: [The utility class name 'PropertyNames' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  4
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  4
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

File: jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java (hunk 3 above)
Comment (splekhanov) [reply jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTest.java]: Done!

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTests.java  2
- jdi-light/src/main/java/com/epam/jdi/light/common/Property.java  1
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

File: jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.epam.jdi.light.common;
+
+import java.security.InvalidParameterException;
+import java.util.Properties;
+
+public class PropertyValidator {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java]: Codacy Issue found: [The utility class name 'PropertyValidator' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

File: jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    public static void validateProperties(Properties properties) {
+        properties.forEach((k, v) -> validate(k.toString(), v.toString()));
+    }
+
+    public static void validate(String name, String value) {
+        for (Property p : Property.values()) {
+            if (p.getName().equals(name)) {
+                if (!value.matches(p.getCheck())) {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTests.java  2
- jdi-light/src/main/java/com/epam/jdi/light/common/Property.java  1
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

File: jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.epam.jdi.light.common;
+
+import java.security.InvalidParameterException;
+import java.util.Properties;
+
+public class PropertyValidator {
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidator.java]: Codacy Issue found: [The utility class name 'PropertyValidator' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper)'](https://app.codacy.com/app/42_/jdi-light/pullRequest?prid=4532609)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidationUtils.java  5
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTests.java  2
- jdi-light/src/main/java/com/epam/jdi/light/common/Property.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light/src/main/java/com/epam/jdi/light/common/PropertyValidationUtils.java  5
- jdi-light/src/test/java/io/github/epam/properties/TimeoutWaitElementTests.java  2
- jdi-light/src/main/java/com/epam/jdi/light/common/Property.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/jdi-light/jdi-light/pullRequest?prid=4532609&bid=15256239)

**Comment by RomE-HK** [comment]
@splekhanov you don't to need to add Codacy to reviewers manually with each new commit =)

**Pull request merged**